solc/CommandLineParser.cpp

```diff
@@ -283,6 +283,95 @@ bool checkMutuallyExclusive(boost::program_options::variables_map const& args, s
 
 }
 
+bool OutputSelection::operator==(OutputSelection const& _other) const
+{
+	static_assert(
+		sizeof(*this) == 15 * sizeof(bool),
+		"Remember to update code below if you add/remove fields."
+	);
+
+	return
+		astCompactJson == _other.astCompactJson &&
+		asm_ == _other.asm_ &&
+		asmJson == _other.asmJson &&
+		opcodes == _other.opcodes &&
+		binary == _other.binary &&
+		binaryRuntime == _other.binaryRuntime &&
+		abi == _other.abi &&
+		ir == _other.ir &&
+		irOptimized == _other.irOptimized &&
+		ewasm == _other.ewasm &&
+		signatureHashes == _other.signatureHashes &&
+		natspecUser == _other.natspecUser &&
+		natspecDev == _other.natspecDev &&
+		metadata == _other.metadata &&
+		storageLayout == _other.storageLayout;
+}
+
+bool CombinedJsonRequests::operator==(CombinedJsonRequests const& _other) const
+{
+	static_assert(
+		sizeof(*this) == 17 * sizeof(bool),
+		"Remember to update code below if you add/remove fields."
+	);
+
+	return
+		abi == _other.abi &&
+		metadata == _other.metadata &&
+		binary == _other.binary &&
+		binaryRuntime == _other.binaryRuntime &&
+		opcodes == _other.opcodes &&
+		asm_ == _other.asm_ &&
+		storageLayout == _other.storageLayout &&
+		generatedSources == _other.generatedSources &&
+		generatedSourcesRuntime == _other.generatedSourcesRuntime &&
+		srcMap == _other.srcMap &&
+		srcMapRuntime == _other.srcMapRuntime &&
+		funDebug == _other.funDebug &&
+		funDebugRuntime == _other.funDebugRuntime &&
+		signatureHashes == _other.signatureHashes &&
+		natspecDev == _other.natspecDev &&
+		natspecUser == _other.natspecUser &&
+		ast == _other.ast;
+}
+
+bool CommandLineOptions::operator==(CommandLineOptions const& _other) const
+{
+	return
+		sourceFilePaths == _other.sourceFilePaths &&
```

Comment:
Do we really need this equality operator? Isn't it very dangerous in case we forget to add some new setting here?

Reply:
The alternative is comparing individual properties one by one, with the same risk.

I thought about adding `static_assert` for the size of the struct like I did in operators for the structs containing booleans but the assert was almost as big as the comparison itself so I thought it was too much. It would decrease the risk a lot though.

Reply:
Possible using something like here could help? 

https://stackoverflow.com/questions/17660095/iterating-over-a-struct-in-c

Reply:
I see that solutions boil down to writing a bunch of custom macros or using [Boost.Fusion](https://www.boost.org/doc/libs/1_76_0/libs/fusion/doc/html/fusion/quick_start.html) (which is also a bunch of macros and templates :)). I'd be fine with Boost.Fusion - looks like it would solve the problem of operators potentially going out of sync and usage is not complicated. Still, it adds a bunch of weird macro calls and I'd still have to repeat the list of members once (in macro invocation). But that's not a bad trade-off in my opinion. @chriseth What's your opinion on using Boost.Fusion?

Reply:
Not sure about that, especially since we already have the code here. I also heard that fusion is a beast. About going out of sync: Would an assert using sizeof not be enough?

Reply:
I think the assert is good enough but this particular function does not have one. I'd have to list all field types from `CommandLineOptions` in it and there's a lot of them. I considered adding it but in the end I thought that an assert the size of the whole function might be an overkill :)

Also, C++20 will provide default equality operators which would be a much cleaner solution.